scripts/benchmark_multithreaded.py

```diff
@@ -29,7 +29,7 @@
 
```

Comment:
(random line)
@mrks What about introducing a "report_format_version" attribute in the report context, given that we'll probably keep on changing the format of the JSON-report. With this, a BenchmarkCI could remain compatible with old commits as well, when newer have a new JSON-report version.

Reply:
I thought about that for a second, but the time where we compare different versions is probably limited. Having to maintain multiple versions of something is generally a pain. I believe it would be easier to hack the compare_benchmark script accordingly if needed.

Reply:
If we ever want to re-benchmark older versions of hyrise we're pretty much lost if we do not have the report-version. Just add it, doesn't hurt if it's there?

Reply:
Mainly thinking about a potential benchmark CI/speed center here

Reply:
Ah, in the C++ code, you just want to add an identifier. Sure, we can do that. I first thought that you wanted to emit both versions.

Reply:
correct, I want the speedcenter to deal with multiple JSON-report versions/layouts, if anyone has to.

Reply:
Any idea how to enforce the correctness of this value? Until we have a user of this value, I foresee situations where the JSON is changed but the value is not. Some time later, there would be a "Oups, forgot it" commit; so instead of two versions, we now have three. If that happens, the JSON user would have to identify the version by the presence / absence of fields instead of the version ID.

Reply:
Not really, tbh, we would just have to look out for it, I suppose... :( But given the infrequency that changes to the JSON format would happen, we would hopefully catch it in reviews?

Reply:
Given the infrequency of those changes, I would expect people to have already forgotten about this (if they knew about it in the first place) ;)

Reply:
Basic testing for the JSON format could pretty easily be integrated into #1103. I'd add the `report_version`. Thank me in a year or so, when you can look at the shiny performance development of Hyrise, beautifully plotted over the course of months and years, not just the last few commit until the BenchmarkCI ran into a JSON that it didn't understand.

Reply:
Wait a second - we already *have* a version identifier, which is guaranteed to be always up-to-date: Why don't we use the git hash?

Reply:
Because we don't know all the git hashes that use a particular JSON format.

Reply:
We don't need to. If we change the format at a certain point, we just need to tell the consumer that "from this hash on, you need to xyz".

Reply:
Mmh, I guess that would work....